implementations/rust/ockam/ockam_api/src/cloud/share/create.rs

```diff
@@ -0,0 +1,163 @@
+use minicbor::{Decode, Encode};
+use serde::Serialize;
+
+use ockam_identity::IdentityIdentifier;
+
+use super::{RoleInShare, SentInvitation, ShareScope};
+
+#[derive(Clone, Debug, Decode, Encode, Serialize)]
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct CreateInvite {
+    #[n(1)] pub expires_at: Option<String>,
+    #[n(2)] pub grant_role: RoleInShare,
+    #[n(3)] pub recipient_email: Option<String>,
+    #[n(4)] pub remaining_uses: Option<usize>,
+    #[n(5)] pub scope: ShareScope,
+    #[n(6)] pub target_id: String,
+}
+
+#[derive(Clone, Debug, Decode, Encode, Serialize)]
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct CreateServiceInvite {
+    #[n(1)] pub expires_at: Option<String>,
+    #[n(2)] pub project_id: String,
+    #[n(3)] pub recipient_email: String,
+
+    // TODO: Is there an identity type I should use that may be CBOR-encoded?
+    // TODO: Should route be a MultiAddr?
+    #[n(4)] pub project_identity: IdentityIdentifier,
+    #[n(5)] pub project_route: String,
```

Comment:
We can. MultiAddr has a `try_from_cbor` method, but I'm sure if we should implement the Decode/Encode traits manually for it to work though. We haven't used this anywhere yet, so we can investigate this later and refactor more requests types.

Reply:
I would like to pursue that in a future PR for correctness' sake, but I don't think it's mechanically needed for this behavior for now. The backend is currently going to store and retrieve the route fields as an opaque string value and pass it back to clients unaltered, without validating its syntax or semantics.